catroid/src/androidTestPhiro/java/org/catrobat/catroid/uiespresso/ui/activity/PhiroMyProjectsActivityTest.java

```diff
@@ -0,0 +1,148 @@
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.bricks.Brick;
+import org.catrobat.catroid.content.bricks.NoteBrick;
+import org.catrobat.catroid.exceptions.CompatibilityProjectException;
+import org.catrobat.catroid.exceptions.LoadingProjectException;
+import org.catrobat.catroid.exceptions.OutdatedVersionProjectException;
+import org.catrobat.catroid.io.StorageHandler;
+import org.catrobat.catroid.phiro.ui.PhiroMainMenuActivity;
+import org.catrobat.catroid.uiespresso.util.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static junit.framework.Assert.assertTrue;
+
+import static org.catrobat.catroid.uiespresso.util.matchers.AdapterViewMatchers.withAdaptedDataAt;
+
+@RunWith(AndroidJUnit4.class)
+public class PhiroMyProjectsActivityTest {
+
+	private static final String TAG = PhiroMyProjectsActivityTest.class.getSimpleName();
+	@Rule
+	public BaseActivityInstrumentationRule<PhiroMainMenuActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(PhiroMainMenuActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		removePhiroPrograms();
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void phiroProgramsInitiallyLoadedTest() {
+		onView(withText(R.string.main_menu_programs))
+				.perform(click());
+
+		ViewInteraction viewInteraction = onView(withId(android.R.id.list));
+
+		Integer position = 0;
+		for (String program : PhiroMainMenuActivity.phiroProjects) {
+			viewInteraction
+					.check(matches(withAdaptedDataAt(program, ++position)));
+		}
+	}
+
+	@Test
+	public void phiroProgramsRestoredWhenDeletedTest() {
+		onView(withText(R.string.main_menu_programs))
+				.perform(click());
+
+		StorageHandler.getInstance().deleteProject(PhiroMainMenuActivity.phiroProjects[0]);
+
+		onView(withId(android.R.id.home))
+				.perform(click());
+
+		onView(withText(R.string.main_menu_programs))
+				.perform(click());
+
+		onView(withId(android.R.id.list))
+				.check(matches(withAdaptedDataAt(PhiroMainMenuActivity.phiroProjects[0], 0)));
+	}
+
+	@Test
+	public void phiroProgramsUnchangedWhenModifiedTest() {
+		onView(withText(R.string.main_menu_programs))
+				.perform(click());
+
+		addBrickToProject(PhiroMainMenuActivity.phiroProjects[0]);
+
+		onView(withId(android.R.id.home))
+				.perform(click());
+
+		onView(withText(R.string.main_menu_programs))
+				.perform(click());
+
+		checkBrickInProject();
+	}
+
+	private void addBrickToProject(String projectName) {
```

Comment:
This tests the following scenario:
1. An existing Phiro program is changed
2. User navigates back to MainMenu
3. User navigates to ProjectList
4. The changed Phiro program should be in the same state (not overridden)